src/opengl/renderer.cc

```diff
@@ -269,39 +274,198 @@ void RendererProgram<camera::FisheyePolynomialTangentialCamera>::SetUniformValue
+         "} else {\n"
+         "  r2 = 99.0;\n"
+         "}\n"
+         "localPoint.x = localPoint.x * r2;\n"
+         "localPoint.y = localPoint.y * r2;\n";
+}
+
+void RendererProgram<camera::RadialCamera>::GetUniformLocations(
+    const ShaderProgramOpenGL& shader_program) {
+  u_k1_location_ = shader_program.GetUniformLocationOrAbort("k1");
+  u_k2_location_ = shader_program.GetUniformLocationOrAbort("k2");
+  radius_cutoff_squared_location_ = shader_program.GetUniformLocationOrAbort("radius_cutoff_squared");
+}
+
+void RendererProgram<camera::RadialCamera>::SetUniformValues(
+    const camera::RadialCamera& camera) const {
+  glUniform1f(u_k1_location_, camera.distortion_parameters().x());
+  glUniform1f(u_k2_location_, camera.distortion_parameters().y());
+  glUniform1f(radius_cutoff_squared_location_, camera.radius_cutoff_squared());
+}
+
+const GLchar* RendererProgram<
+    camera::RadialFisheyeCamera>::GetShaderUniformDefinitions() const {
+  return "uniform float k1;\n"
+         "uniform float k2;\n"
+         "uniform float radius_cutoff_squared;\n";
+}
+
+const GLchar*
+RendererProgram<camera::RadialFisheyeCamera>::GetShaderDistortionCode() const {
+  // (Mis)using localPoint.w for intermediate results.
+  return "float nx = localPoint.x / localPoint.z;\n"
+         "float ny = localPoint.y / localPoint.z;\n"
+         "float r2 = nx * nx + ny * ny;\n"
+         "float r = sqrt(r2);\n"
+         "if (r > 1e-6) {\n"
+         "  float theta_by_r = atan(r, 1.0) / r;\n"
+         "  nx = theta_by_r * nx;"
+         "  ny = theta_by_r * ny;"
+         "  r2 = nx * nx + ny * ny;"
+         "}\n"
+         "if (r2 <= radius_cutoff_squared) {\n"
+         "  r2 = 1.0 + r2 * (k1 + r2 * k2);\n"
+         "} else {\n"
+         "  r2 = 99.0;\n"
+         "}\n"
+         "localPoint.x = localPoint.z * r2 * nx;\n"
+         "localPoint.y = localPoint.z * r2 * ny;\n";
+}
+
+void RendererProgram<camera::RadialFisheyeCamera>::GetUniformLocations(
+    const ShaderProgramOpenGL& shader_program) {
+  u_k1_location_ = shader_program.GetUniformLocationOrAbort("k1");
+  u_k2_location_ = shader_program.GetUniformLocationOrAbort("k2");
+  radius_cutoff_squared_location_ = shader_program.GetUniformLocationOrAbort("radius_cutoff_squared");
+}
+
+void RendererProgram<camera::RadialFisheyeCamera>::SetUniformValues(
+    const camera::RadialFisheyeCamera& camera) const {
+  glUniform1f(u_k1_location_, camera.distortion_parameters().x());
+  glUniform1f(u_k2_location_, camera.distortion_parameters().y());
+  glUniform1f(radius_cutoff_squared_location_, camera.radius_cutoff_squared());
+}
+
+const GLchar* RendererProgram<
+    camera::SimpleRadialCamera>::GetShaderUniformDefinitions() const {
+  return "uniform float k;\n"
+         "uniform float radius_cutoff_squared;\n";
+}
+
+const GLchar*
+RendererProgram<camera::SimpleRadialCamera>::GetShaderDistortionCode() const {
+  // (Mis)using localPoint.w for intermediate results.
```

Comment:
Comment seems outdated

Reply:
Speaking of which, is using an already declared variable (here localPoint.x) better for speed in opengl ?
I cleaned the code by using a dedicated variable wih its own name, but i wonder if it was at a cost of a slower execution.

Reply:
Actually I have no idea ;) I think the original implementation was just being overly cautious. If it works to use a separate variable then this cleaner implementation clearly seems preferable.